test/test_reactant.jl

```diff
@@ -1,15 +1,50 @@
 using Reactant
+
 if haskey(ENV, "GPU_TEST")
     Reactant.set_default_backend("gpu")
 else
     Reactant.set_default_backend("cpu")
 end
+
 using Test
 using Oceananigans
-using Reactant
 using Oceananigans.Architectures
+using Oceananigans.BoundaryConditions: fill_halo_regions!
 using GPUArrays
-GPUArrays.allowscalar(true)
+
+@testset "Reactanigans unit tests" begin
+    arch = ReactantState()
+    grid = RectilinearGrid(arch; size=(4, 4, 4), extent=(1, 1, 1))
+    c = CenterField(grid)
+    @test parent(c) isa Reactant.ConcreteRArray
+
+    set!(c, (x, y, z) -> x + y * z)
+    x, y, z = nodes(c)
+
+    @allowscalar begin
+        @test c[1, 1, 1] == x[1] + y[1] * z[1]
+        @test c[1, 2, 1] == x[1] + y[2] * z[1]
+        @test c[1, 2, 3] == x[1] + y[2] * z[3]
+    end
+
+    fill_halo_regions!(c)
```

Comment:
you should likely do `@jit fill_halo_regions!(c)` here (which is equiv to compile and then run).